### DISCUSSION TITLE
Problem with hombrew install

### DISCUSSION THREAD

**Discussion by dectun** [Homebrew/discussions#3175] [Getting started]
Hello. I can´t install MVE, and I review the steps to install homebrew. When I apply the brew doctor command it get me this;
![Captura de Pantalla 2022-04-12 a las 17 14 35](https://user-images.githubusercontent.com/103385292/162997883-077cf715-50db-4365-b756-9afc0ece766d.png)

really, I need this software for a paleontologic proyect of University of Extremadura. And I need your help. Thanks.

**Comment by SMillerDev** [comment]
Can you please copy the text into the issue and report the output of all commands you tried? Why doesn't `mve` work?

**Comment by dectun** [comment]
Ok, I explain all steps that I did.


I used this link;
(https://github.com/simonfuhrmann/mve/wiki/Build-Instructions-for-OS-X)

I installed the XCODE, all packet. Then I installed de macports. Because One person said me this is necessary.
Then I installed de hombrew. When I try to put the command to install the homebrew I haven`t problem. And with the clone MVE repossitory it`s good.

![Captura de Pantalla 2022-04-12 a las 18 04 58](https://user-images.githubusercontent.com/103385292/163006216-c29aaec7-8bdf-45b2-a117-a9ca003e83aa.png)


But then, when I try to put the build MVE and UMVE step. I had a problem like this; clang: error: the clang compiler does not support '-march=native'. For this, one person said me that I need change the code of Makefile.inc. And I change this. In this step, I try repair the code of system.h in 149 line. Like issue 533 said. And I put in this line; 
![Captura de Pantalla 2022-04-12 a las 5 45 25](https://user-images.githubusercontent.com/103385292/163006861-dfc4eb5e-a4f6-45ca-b433-c97118fbe899.png)

![Captura de Pantalla 2022-04-12 a las 18 12 44](https://user-images.githubusercontent.com/103385292/163007108-b35c0d24-1c2d-491a-ad2d-4f89d96b8cc4.png)


And then. I try with suggest of simonfuhrmann, with this link;
[
](https://github.com/simonfuhrmann/mve#building-mve-and-umve)

The first and second steps was rigth, but when I try put the "make -j8" command it get me this; [
](https://github.com/simonfuhrmann/mve#building-mve-and-umve)


At last. I try this two command: "sudo find / -name png.h" and "brew doctor". And I get this:
![Captura de Pantalla 2022-04-12 a las 18 16 49](https://user-images.githubusercontent.com/103385292/163008261-f0ec434e-b5c6-4cd7-a3a5-9d5606e13ad7.png)


I don`t be sure if this is that you need. I haven`t good english. So sorry for this.

**Reply by dectun** [reply]
sorry, the link of simonfuhrmann is eraser from the web. This is a screemshot of the page.
![Captura de Pantalla 2022-04-12 a las 18 23 17](https://user-images.githubusercontent.com/103385292/163009124-6e100b48-0bbf-4419-9922-b834ab375917.png)

**Reply by SMillerDev** [reply]
> Then I installed de macports. Because One person said me this is necessary.

Macports and homebrew fulfill the same purpose so this is not needed. If anything it'll make problems harder to debug since you'll have two locations to look in.

---

So it seems you're trying to install something on an M1 Mac using a guide written largely  before M1 existed. Did you follow the instructions here: https://github.com/simonfuhrmann/mve/wiki/Build-Instructions-for-OS-X#for-apple-m1-chip-users?

**Reply by dectun** [reply]
ok. I think, that I understand the problem. But, I don`t know what is the best new step. do you think the best new step is uninstall macports?

**Reply by dectun** [reply]
I uninstall macport. brew doctor said "system is ready to brew".

I try with these commands. But the error of clang compiler is continuos.
![Captura de Pantalla 2022-04-12 a las 21 01 06](https://user-images.githubusercontent.com/103385292/163034955-71967186-797f-4405-bb4e-d854bfa07211.png)

**Reply by SMillerDev** [reply]
If it works with those instructions you don't need to do anything. If it is still broken removing Macports would make the dependencies easier to debug.

**Reply by dectun** [reply]
ok. I removed all of macports. Now, there is only one bug; clang: clangerror: the clang compiler does not support '-march=native':

before, I changed the command of -march=native to -mcpu=apple-m1. Like the screemshot. But I don`t know if I did in a wrong way.
![Captura de Pantalla 2022-04-12 a las 5 45 25](https://user-images.githubusercontent.com/103385292/163038602-138a6b05-cafe-4b3f-bd80-91d1020636be.png)
.

**Reply by dectun** [reply]
Ok, I found three makefile.inc. I rewrite all with this command; -mcpu=apple-m1. 
Now, when I try with these command; "make -j8!" or "make", it get me: ../../libs/util/system.h:245:5: error: "Couldn't determine host endianess!".

In my systme.h file of MVE. I have that is on the screemshot. is it the correcto code?
![Captura de Pantalla 2022-04-12 a las 22 23 43](https://user-images.githubusercontent.com/103385292/163047489-0d4666fa-0ad8-45f0-a517-9558386e4fcc.png)

**Reply by SMillerDev** [reply]
It's in the instructions I linked. It needs `__aarch64__` at the end. The `__arm__` you added doesn't actually do anything do you can remove that.

**Reply by dectun** [reply]
Ok, I did it. but the bug stay. like the screemshot.
![Captura de Pantalla 2022-04-13 a las 3 53 56](https://user-images.githubusercontent.com/103385292/163084059-383555bb-fa30-4532-881d-31b5d1388a31.png)


One person said me, that in the chip M1 for Mac need defined if is LE or BE. The reason that he said, is this chip are similar to ARM.
And I don`t understand what say this note in the file.
![Captura de Pantalla 2022-04-13 a las 3 51 49](https://user-images.githubusercontent.com/103385292/163084516-9271dbe1-edbb-446b-9ad0-bfdd67b9420c.png)

**Reply by dectun** [reply]
hello. good morning in Spain. 

I think, need defined the arquitecture in the position of (HOST_BYTEORDER_BE) at line 147.

 is possible put de code "|| defined(__aarch64__)" on the line 147?.
![Captura de Pantalla 2022-04-13 a las 8 42 44](https://user-images.githubusercontent.com/103385292/163116600-51ef0953-ee5b-437e-8002-f214385537a4.png)

**Comment by dectun** [comment]
I put here, all rest command that I used during the installation. In screenshots.
![Captura de Pantalla 2022-04-12 a las 19 13 53](https://user-images.githubusercontent.com/103385292/163018707-ea593038-300b-4599-9341-e383701c3a51.png)
![Captura de Pantalla 2022-04-12 a las 19 16 55](https://user-images.githubusercontent.com/103385292/163018709-df06539d-1234-4377-88d6-2ac3af12053f.png)
![Captura de Pantalla 2022-04-12 a las 19 16 34](https://user-images.githubusercontent.com/103385292/163018711-a1ef2c07-3226-40f2-8b16-8bb174c07f5d.png)
![Captura de Pantalla 2022-04-12 a las 19 16 12](https://user-images.githubusercontent.com/103385292/163018712-41d3da10-c19a-4630-b9ce-8c9def3bf740.png)
![Captura de Pantalla 2022-04-12 a las 19 15 49](https://user-images.githubusercontent.com/103385292/163018717-66260c9d-47a7-41d3-b59a-77272e963ba0.png)
![Captura de Pantalla 2022-04-12 a las 19 15 39](https://user-images.githubusercontent.com/103385292/163018720-c0d1b72e-b76b-4cd0-b5e3-92d44f0c7be5.png)
![Captura de Pantalla 2022-04-12 a las 19 15 17](https://user-images.githubusercontent.com/103385292/163018722-27e6394e-cd31-493f-87b6-93f138e32ed1.png)
![Captura de Pantalla 2022-04-12 a las 19 14 59](https://user-images.githubusercontent.com/103385292/163018723-bf7142a6-9c73-4b73-9618-5a08212de43e.png)
![Captura de Pantalla 2022-04-12 a las 19 14 34](https://user-images.githubusercontent.com/103385292/163018727-31878d6a-00cd-45bd-b6ae-1c8c4c785066.png)
![Captura de Pantalla 2022-04-12 a las 19 14 20](https://user-images.githubusercontent.com/103385292/163018728-4516a015-d448-4bd3-bc23-a8e69a114f1a.png)
![Captura de Pantalla 2022-04-12 a las 19 14 04](https://user-images.githubusercontent.com/103385292/163018730-c64252c8-9787-4818-a994-5aa284d2c4d4.png)
